ops-catalog/generate-migration.sh

```diff
@@ -19,36 +19,118 @@ if [[ -z "$ENVIRONMENT" ]]; then
 	ops_user_id uuid;
 	current_tenant tenants;
+	tenant_count integer;
+	current_l1_stat_rollup integer;
+	tenant_spec jsonb;
+	l1_derivation_specs jsonb = '{}';
+	l2_derivation_spec jsonb;
+	materialization_spec jsonb;
 begin
 
 	-- Identify user which owns ops specifications.
 	select id into strict ops_user_id from auth.users where email = 'support@estuary.dev';
 
-	-- Update the ops catalog template.
-	update ops_catalog_template set bundled_catalog = bundled_catalog_arg;
+	-- Update the ops catalog templates.
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_TENANT}' where template_type = 'tenant';
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_LEVEL_1_DERIVATION}' where template_type = 'l1_derivation';
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_LEVEL_2_DERIVATION}' where template_type = 'l2_derivation';
+	update ops_catalog_template set bundled_catalog = '${BUNDLED_MATERIALIZATION}' where template_type = 'materialization';
+
+	-- There must be at least one tenant in the system to proceed. This is a guard for local
+	-- development when first starting the local stack.
+	select count(*)
+	into strict tenant_count
+	from tenants;
+	if tenant_count = 0 then
+		return;
+	end if;
+
+	-- Create a draft of ops changes.
+	insert into drafts (id, user_id, detail) values
+	(new_draft_id, ops_user_id, 're-publishing ops catalog');
+
+	-- Queue a publication of the draft.
+	insert into publications (id, user_id, draft_id) values
+	(publication_id, ops_user_id, new_draft_id);
 
-	-- Start a publication of the updated template for each existing tenant.
+	-- Handle the tenant-specific specs for each tenant.
 	for current_tenant in
 		select * from tenants
 	loop
-		perform internal.create_ops_publication(current_tenant.tenant, ops_user_id);
+		select replace(bundled_catalog::text, 'TENANT', rtrim(current_tenant.tenant, '/'))::jsonb
+		into strict tenant_spec
+		from ops_catalog_template
+		where template_type = 'tenant';
+
+		-- Draft a deletion for all currently-live ops catalog specs for this tenant.
+		insert into draft_specs (draft_id, catalog_name, spec_type, spec)
+		select new_draft_id, catalog_name, null, null
+		from live_specs
+		where catalog_name like ('ops/' || current_tenant.tenant || '%');
+
+		-- Upsert the tenant-specific draft specs.
+		insert into draft_specs (draft_id, catalog_name, spec_type, spec)
+		select new_draft_id, "key", 'capture'::catalog_spec_type, "value"
+		from jsonb_each(jsonb_extract_path(tenant_spec, 'captures'))
+		union all
+		select new_draft_id, "key", 'collection'::catalog_spec_type, "value"
+		from jsonb_each(jsonb_extract_path(tenant_spec, 'collections'))
+		union all
+		select new_draft_id, "key", 'materialization'::catalog_spec_type, "value"
+		from jsonb_each(jsonb_extract_path(tenant_spec, 'materializations'))
+		on conflict (draft_id, catalog_name)
+		do update set spec_type = excluded.spec_type, spec = excluded.spec;
```

Comment:
Here and in the `create_ops_publication` postgres functions I have opted not to include the tests for each tenant. This is partially an attempt to simplify things, and also because I was concerned about a new tenant publication needing to run the tests of every single other tenant attached to that level 1 derivation. I could be convinced to include them though.